### PR TITLE
feat: `fetch` API

### PR DESCRIPTION
**Pull request by tusharmath**
**Summary:**  
_Briefly describe the changes made in this PR._

**Issue Reference(s):**  
Fixes #... _(Replace "..." with the issue number)_

**Build & Testing:**

- [ ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

**Comment by tusharmath** [comment]
Start the server with — 
```
cargo run -- start ./examples/jsonplaceholder_script.graphql 
```

File: src/cli/javascript/worker.rs [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                    .clone();
+
+                Ok(v8.as_sync_function(closure))
+            })
+            .await?;
+        Ok(Self { sync_v8: sync_v8.clone(), on_event_js: closure })
+    }
+
+    pub async fn on_event(&self, request: reqwest::Request) -> anyhow::Result<Response<Bytes>> {
+        let (tx, mut rx) = tokio::sync::broadcast::channel::<serde_json::Value>(1024);
```
Comment (meskill) [inline src/cli/javascript/worker.rs:49]: why using broadcast channel? it's many to many channel. Probably, oneshot is enough for single

Comment (tusharmath) [reply src/cli/javascript/worker.rs:49]: We need to await asynchronously, i don't think its possible with one shot. Will check again.

**Comment by tusharmath** [comment]
Main learnings from this PR are —

- Serde takes up much time sending messages from JS to Tailcall and vice-versa.
- Performance improvements can be made by using serde_v8 crate
- There is a lot of unsafe usage that causes segmentation faults.

Closing this PR.

**Pull request closed without merging**